src/storage-manager.coffee

```diff
@@ -67,6 +61,9 @@ class _StorageManager
     collection.on 'remove', (model) ->
       model.invalidateCache()
 
+    collection.on 'change', (model) =>
+      @_invalidateCache(model.brainstemKey) unless model.isNew()
```

Comment:
I am thinking this is a good approximation that is hooked into vanilla Backbone. There is a possible solution where the `save` methods are overridden to emit an extra event (i.e. `saved`). I was thinking that could be an iteration after seeing the effects of the change.